src/Subscriber/CategoryPageSubscriber.php

```diff
@@ -52,12 +58,23 @@ public function onPageLoaded(NavigationPageLoadedEvent $event): void
 
         $communication = [
                 'search-immediate' => !$isHome && !$disableImmediate ? 'true' : 'false',
-                'add-params'       => implode(',', $this->addParams),
+                'add-params'       => implode(',', $this->prepareParameters($this->categoryPageAddParams) + $this->prepareParameters($this->configurationAddParams)),
             ] + ($isCategory ? ['category-page' => $this->prepareCategoryPath($categoryPath)] : []);
 
         $event->getPage()->getExtension('factfinder')->assign(['communication' => $communication]);
     }
 
+    private function prepareParameters(array $parametersArray): array
+    {
+        $result = [];
+
+        foreach ($parametersArray as $value) {
```

Comment:
There is lot of repeated explode on the same value. This could be definitely reworked.
I always find operating on numerical indexes of arrays risky and more difficult to read.

why not use reduce to transform array of key=value expressions to the an associative array which will guarantee we would not have duplicated parameter names


        $transformed = array_reduce($this->addParams + $baseAddParams, function (array $acc, string $expr) {
            list($key, $value) = explode('=', $expr);
            return $acc + [$key => $value];
        }, []);

and then when assigning

       'add-params'       =>  implode(',', array_map(fn(string $key, string $value) => sprintf('%s=%s', $key, $value), array_keys($mergedAddParams), array_values($mergedAddParams))),

Reply:
Thanks for the advice, this solution looks better